lib/handlers/extractHandler.ts

```diff
@@ -85,46 +83,99 @@ export class StagehandExtractHandler {
 
     await this.waitForSettledDom(domSettleTimeoutMs);
     await this.startDomDebug();
-    const { outputString, chunk, chunks } = await this.stagehand.page.evaluate(
-      (chunksSeen?: number[]) => window.processDom(chunksSeen ?? []),
-      chunksSeen,
-    );
+
+    const originalDOM = await this.stagehand.page.evaluate(() => window.storeDOM());
+
+    const { selectorMap }: { selectorMap: Record<number, string[]> } =
+      await this.stagehand.page.evaluate(() => window.processAllOfDom());
 
     this.logger({
       category: "extraction",
-      message: "received output from processDom.",
-      auxiliary: {
-        chunk: {
-          value: chunk.toString(),
-          type: "integer",
-        },
-        chunks_left: {
-          value: (chunks.length - chunksSeen.length).toString(),
-          type: "integer",
-        },
-        chunks_total: {
-          value: chunks.length.toString(),
-          type: "integer",
-        },
-      },
+      message: `received output from processAllOfDom. selectorMap has ${Object.keys(selectorMap).length} entries`,
+      level: 1,
     });
+    const PROXIMITY_THRESHOLD = 10;
+    await this.stagehand.page.evaluate(() => window.createTextBoundingBoxes());
+    const pageWidth = await this.stagehand.page.evaluate(() => window.innerWidth);
+    const pageHeight = await this.stagehand.page.evaluate(() => window.innerHeight);
+
+    const seenAnnotations = new Map();
+    const textAnnotations: TextAnnotation[] = [];
+
+    for (const xpaths of Object.values(selectorMap)) {
+      const xpath = xpaths[0];
+
+      const boundingBoxes: Array<{
+        text: string;
+        left: number;
+        top: number;
+        width: number;
+        height: number;
+      }> = await this.stagehand.page.evaluate(
+        (xpath) => window.getElementBoundingBoxes(xpath),
+        xpath
+      );
+
+      for (const box of boundingBoxes) {
+        const text = box.text;
+
+        let annotationsForText = seenAnnotations.get(text);
+        if (!annotationsForText) {
+          annotationsForText = [];
+          seenAnnotations.set(text, annotationsForText);
+        }
+
+        const isDuplicate = annotationsForText.some((annotation) => {
+          const dx = annotation.x - (box.left + box.width / 2);
+          const dy = annotation.y - (box.top + box.height / 2);
+          const distance = Math.sqrt(dx * dx + dy * dy);
+          return distance < PROXIMITY_THRESHOLD;
```

Comment:
Check for deduping by checking bounding box + same word; this works but doesn't get rid of all of them. Not sure why but there are situations where it'll fail (won't remove duplicates or will still add). 

Thus, add proximity threshold to give you a tolerance for deduping. This is hacky. Is "10" 10px?